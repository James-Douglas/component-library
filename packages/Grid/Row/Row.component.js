import React from 'react';
import PropTypes from 'prop-types';
import StyledFlexRow from './Row.styles';

const Row = ({
  children,
  className,
  reverse,
  removeMarginBottom,
  flexWrap,
  justify,
}) => (
  <StyledFlexRow
    justify={justify}
    className={`row ${className}`}
    reverse={reverse}
    removeMarginBottom={removeMarginBottom}
    flexWrap={flexWrap}
  >
    {children}
  </StyledFlexRow>
);

Row.propTypes = {
  /**
   * Classes to be applied to the row element
   */
  className: PropTypes.string,
  /**
   * Flex-direction: reverse bool for the row
   */
  reverse: PropTypes.bool,
  /**
   * The children of the component
   */
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.object,
  ]),
  /**
   * Remove the default margin-bottom
   */
  removeMarginBottom: PropTypes.bool,
  /**
   * It defines whether the flex items are forced in a single line or can be flowed into multiple lines.
   */
  flexWrap: PropTypes.oneOf([
    'nowrap',
    'wrap',
    'wrap-reverse',
  ]),
  /**
   * Defines alignment of a column
   */
  justify: PropTypes.oneOf([
    'flex-start',
    'center',
    'flex-end',
    'space-around',
    'space-between',
    'initial',
    'inherit',
  ]),
};

Row.defaultProps = {
  className: '',
  children: [],
  removeMarginBottom: false,
  reverse: false,
  flexWrap: 'wrap',
  justify: 'flex-start',
};

export default Row;
