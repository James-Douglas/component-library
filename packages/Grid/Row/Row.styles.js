import styled, { css } from 'styled-components';
import { Row as FlexRow } from 'react-styled-flexboxgrid';

const StyledFlexRow = styled(FlexRow)`
  box-sizing: border-box;
  margin-bottom: ${({ theme, removeMarginBottom }) => (removeMarginBottom ? 0 : theme.spacing[24])};
  ${(props) => (props.flexWrap) && css`
    flex-wrap: ${props.flexWrap};
  `};
  ${(props) => (props.reverse) && css`
    flex-direction: row-reverse;
  `};
  justify-content: ${(props) => props.justify};
`;

export default StyledFlexRow;
