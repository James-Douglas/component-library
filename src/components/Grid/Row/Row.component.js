import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

const StyledRow = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-left: -1.6rem;
  margin-right: -1.6rem;
  ${(props) => (props.reverse) && css`
    flex-direction: row-reverse;
  `}
  ${(props) => (props.flexWrap) && css`
    flex-wrap: ${props.flexWrap};
  `}
  margin-bottom: ${({ theme, removeMarginBottom }) => (removeMarginBottom ? 0 : theme.spacing[24])};
  width: 100%;
`;

const Row = ({
  children, className, reverse, removeMarginBottom, flexWrap,
}) => (
  <StyledRow className={`row ${className}`} reverse={reverse} removeMarginBottom={removeMarginBottom} flexWrap={flexWrap}>
    {children}
  </StyledRow>
);

Row.propTypes = {
  /**
   * Classes to be applied to the row element
   */
  className: PropTypes.string,
  /**
   * Flex-direction: reverse bool for the row
   */
  reverse: PropTypes.bool,
  /**
   * The children of the component
   */
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.object,
  ]),
  /**
   * Remove the default margin-bottom
   */
  removeMarginBottom: PropTypes.bool,
  /**
    * It defines whether the flex items are forced in a single line or can be flowed into multiple lines.
  */
  flexWrap: PropTypes.oneOf([
    'nowrap',
    'wrap',
    'wrap-reverse',
  ]),
};

Row.defaultProps = {
  className: '',
  children: [],
  removeMarginBottom: false,
  reverse: false,
  flexWrap: 'wrap',
};

export default Row;
