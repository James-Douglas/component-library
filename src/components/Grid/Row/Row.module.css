.row {
  @apply flex flex-wrap -mx-16;
}
