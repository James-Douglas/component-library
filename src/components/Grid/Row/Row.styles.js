import styled, { css } from 'styled-components';

const StyledRow = styled.div`
display: flex;
flex-wrap: wrap;
margin-left: -${({ theme }) => theme.spacing[16]};
margin-right: -${({ theme }) => theme.spacing[16]};
${(props) => (props.reverse) && css`
  flex-direction: row-reverse;
`}
${(props) => (props.flexWrap) && css`
  flex-wrap: ${props.flexWrap};
`}
margin-bottom: ${({ theme, removeMarginBottom }) => (removeMarginBottom ? 0 : theme.spacing[24])};
width: 100%;
`;

export default StyledRow;
