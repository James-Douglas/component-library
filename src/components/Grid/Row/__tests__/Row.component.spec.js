import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import Row from '../Row.component';

describe('Row', () => {
  it('renders correctly without props', () => {
    const { container } = render(<Row />);
    expect(container).toMatchSnapshot();
  });

  it('renders children', () => {
    const { container } = render(
      <Row>
        <div className="child-content">test</div>
      </Row>,
    );

    const childContent = container.querySelector('.child-content');
    expect(childContent).toBeInTheDocument();
  });

  it('has a row class by default', () => {
    const { container } = render(<Row />);

    const row = container.querySelector('.row');
    expect(row).toBeDefined();
    expect(row).toHaveStyleRule('flex-wrap', 'wrap');
  });

  it('allows additional classes to be passed through', () => {
    const { container } = render(<Row className="test-class" />);

    const test = container.getElementsByClassName('test-class');
    expect(test).toBeDefined();
  });

  it('reverses the flow of the children items', () => {
    const { container } = render(
      <Row className="test-class" reverse>
        <div>One</div>
        <div>two</div>
        <div>three</div>
      </Row>,
    );

    const row = container.querySelector('.test-class');
    expect(row).toHaveStyleRule('flex-direction', 'row-reverse');
  });
});
