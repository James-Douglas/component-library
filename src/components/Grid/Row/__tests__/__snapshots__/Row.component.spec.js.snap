// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders correctly without props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-left: -1.6rem;
  margin-right: -1.6rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-bottom: 2.4rem;
  width: 100%;
}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

@media (min-width:1200px) {

}

<div>
  <div
    class="c0 row "
  />
</div>
`;
